File: 프로그래머스/2/42587. 프로세스/프로세스.go
func solution(priorities []int, location int) int {
	result := 0
	arr := make([][]int, len(priorities))
	for i, val := range priorities {
		if i == location {
			arr[i] = []int{val, 1}
			continue
		}
		arr[i] = []int{val, 0}
	}

	for len(arr) > 0 {
		now := arr[0]
		arr = arr[1:]
		check := false
		for _, val := range arr {
			if now[0] < val[0] {
				check = true
				break
			}
		}
		if check {
			arr = append(arr, now)
			continue
		}
		result++
		if now[1] == 1 {
			return result
		}
	}
	return -1
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    priorities = deque(priorities)
    temp = deque(range(len(priorities)))
    result = 0
    
    while temp:
        now_number = priorities.popleft()
        now_index = temp.popleft()
        pri_check = 1
        
        for i in priorities:
            if now_number < i:
                priorities.append(now_number)
                temp.append(now_index)
                pri_check = 0
                break
        
        if pri_check:
            result += 1
            if now_index == location: return result